lib/grape-swagger/endpoint.rb

```diff
@@ -235,6 +235,11 @@ def parse_request_params(required)
       end
     end
 
+    def param_type_is_array?(param_type)
+      return false unless param_type
+      param_type == 'Array' || param_type =~ /\[\w+\]\z/
+    end
```

Comment:
@dblock … the whole `('[', '[][')` junk could be removed by avoiding the introducing of it here [endpoint.rb#L224](https://github.com/ruby-grape/grape-swagger/blob/master/lib/grape-swagger/endpoint.rb#L224)

I had to make a decision, should the name of the parameter indicating if it is a Hash, Array or not, this solution following the rails naming convention for form data

questions is, is it really needed … think removing it, could made many things easier